### PR TITLE
[Search-based access requests] Include allowed resource IDs in user certs

### PR DESCRIPTION
**Pull request by nklaassen**
This PR extends the implementation for search-based access requests (#10887) to include the list of requested and approved resources in the users SSH and TLS certs. The design is in [RFD 59](https://github.com/gravitational/teleport/blob/master/rfd/0059-search-based-access-requests.md).

Access restrictions to these allowed certs are still not implemented, that will be in a following PR, and will be based on the resource IDs stored in the certs by this PR.

I refactored `TestAccessRequest` in `lib/auth` to a table-driven test based on `testify` rather than `check`, and extended it to test that certs are correctly populated for search-based requests. This test is not meant to exhaustively check access request creation and approval conditions, there is another test for that in `lib/services/access_request_test.go`, this is more of an integration test of actual certificate generation based on access requests.

### CODE REVIEW

**Comment by nklaassen** [comment]
friendly ping @lxea @zmb3

**Review by lxea — APPROVED** [review]

File: lib/auth/access_request_test.go [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+// checkCerts checks that the ssh and tls certs include the given roles, logins,
+// accessRequests, and resourceIDs
+func checkCerts(t *testing.T,
+	certs *proto.Certs,
+	roles []string,
+	logins []string,
+	accessRequests []string,
+	resourceIDs []types.ResourceID,
+) {
+	sshCert, err := sshutils.ParseCertificate(certs.SSH)
```
Comment (zmb3) [inline lib/auth/access_request_test.go:320]: nit: add `t.Helper()` here

File: lib/auth/access_request_test.go [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+	tlsCert, err := tlsca.ParseCertificatePEM(certs.TLS)
+	require.NoError(t, err)
+	tlsIdentity, err := tlsca.FromSubject(tlsCert.Subject, tlsCert.NotAfter)
+	require.NoError(t, err)
+
+	rawSSHCertRoles := sshCert.Permissions.Extensions[teleport.CertExtensionTeleportRoles]
+	sshCertRoles, err := services.UnmarshalCertRoles(rawSSHCertRoles)
+	require.NoError(t, err)
+	sort.Strings(roles)
```
Comment (zmb3) [inline lib/auth/access_request_test.go:331]: Instead of sorting and using `require.Equal`, you can probably get away with `require.ElementsMatch()`.

**Review by zmb3 — APPROVED** [review]

File: lib/auth/access_request_test.go (hunk 2 above)
Comment (nklaassen) [reply lib/auth/access_request_test.go]: Thank you! I was hoping `require.ElementsMatch()` existed and went looking for it, but didn't know the name and couldn't find it. This is much better.

**Pull request merged**